File: CS3315/Assignments/Week3Assignment/topTenHuckFinn.lua
#!/usr/bin/env lua

local freq = {}
for line in io.lines('HuckFinn.txt') do
	local lowerline = string.lower(line)
	for word in string.gmatch(lowerline, "[%a']+") do
		if not freq[word] then
			freq[word] = 1
		else
			freq[word] = freq[word] + 1
		end
	end
end
 
local array = {}
for word, count in pairs(freq) do
	table.insert(array, {word, count})
end
 
table.sort(array, function (a, b) return a[2] > b[2] end)
 
for i = 1, arg[2] or 10 do
	io.write(string.format('%7d %s\n', array[i][2] , array[i][1]))
end

File: CS3315/Assignments/Week3Assignment/topTenHuckFinn.js
#!/usr/bin/env node

var fs = require('fs');
fs.readFile('HuckFinn.txt', 'utf8', function(err, text) {
    if (err) {
        return console.log(err);
    }
    var re = /[a-z']+/gi;
    var words = text.match(re);
    var data = new Map();
    for (word of words) {
        word = word.toLowerCase();
        if (data.has(word)) {
            data.set(word, data.get(word) + 1);
        } else {
            data.set(word, 1);
        }
    }
    var topTen = new Map([...data.entries()].sort((a, b) => b[1] - a[1]));
    var keys = topTen.keys();
    for (var i = 0; i < 10; i++) {
        key = keys.next().value;
        console.log("  ", topTen.get(key), key);
    }
});
